teku/src/main/java/tech/pegasys/teku/Teku.java

```diff
@@ -56,5 +71,24 @@ private static void start(final TekuConfiguration config, final boolean validato
                   System.out.println("Teku is shutting down");
                   node.stop();
                 }));
+    return node;
+  }
+
+  @Override
+  public NodeFacade startFromCLIArgs(String[] cliArgs) {
+    SafeFuture<Node> nodePromise = new SafeFuture<>();
```

Comment:
Using a future here is a bit confusing, I'm pretty sure it actually happens sync so might be better to use an `AtomicReference`.

Reply:
it's still a `SafeFuture`? :)

Reply:
Oh, sorry, forgot about it. Fixed: d06ef8c1